CRM/Utils/Mail/Incoming.php

```diff
@@ -447,11 +449,16 @@ public static function parseAddresses(&$addresses, $token, &$params, &$mail, $cr
    * @return int|null
    */
   public static function getContactID($email, $name, $create, &$mail) {
-    $dao = CRM_Contact_BAO_Contact::matchContactOnEmail($email, 'Individual');
+    $contact = Email::get()
```

Comment:
Does api4 Email.get also handle the multisite like matchContactOnEmail?

And there might be some sense in keeping the prioritization of is_primary that matchContactOnEmail does, e.g. if two Individuals have abc@def.org but one of them has it as primary and the other has it as a secondary, it should probably go with the primary.

Also deleted contacts - it currently doesn't file on those but here it would right?

Reply:
Thanks for your comment @demeritcowboy 
I see now that `matchContactOnEmail` has more logic than I imagined.  Considering all those cases in a query with Email v4 will require some effort.
Do you think that implement the logic doing something like this would be acceptable?:
```php
    $dao = CRM_Contact_BAO_Contact::matchContactOnEmail($email, 'Individual');

    $contactID = NULL;
    if ($dao) {
      $contactID = $dao->contact_id;
    } else {
      $dao = CRM_Contact_BAO_Contact::matchContactOnEmail($email, 'Organization');

      if ($dao) {
        $contactID = $dao->contact_id;
      }
    }
```
Are two queries, I know, but I guess it would still be acceptable on efficiency terms.

Reply:
That seems like it would work.

Reply:
@demeritcowboy ok! updated the code and the PR description